test/test_scatter_gather_ops.py

```diff
@@ -98,7 +98,11 @@ def _test_scatter_base(self, fn, *, device, dtype, is_scalar, reduction,
         if is_scalar:
             src = random.random()
         else:
-            src_size = [random.randint(1, 5) + s for s in idx_size]
+            sign = [1, 1, 1]
+            if src_smaller_than_index:
+                while(min(sign) > 0):
+                    sign = [random.choice([-1, 1]) if s > 1 else 1 for s in idx_size]
```

Comment:
does this mean we don't generate cases where src is smaller than index on dim 0?

Reply:
No, I don't think so, unless I misunderstand your question. The `s` there is the size in a given dimension, so if `s = index.size(d) == 1` for some dimension `d` then it won't generate a `src` tensor that has size less than that along that dimension (since afaik there's no such thing as size `0` in a 3-dimensional tensor and this would break stuff). `s == 1` can only happen on `dim`, though, since all other dimensions in that generator are `>= 10`, so that loop is guaranteed to make at least one of the `src` dimension sizes smaller than the corresponding `index` dimension size.

Reply:
My bad, gotcha thanks!